modules/game_market/marketprotocol.lua

```diff
@@ -68,10 +85,20 @@ local function parseMarketLeave(protocol, msg)
 end
 
 local function parseMarketDetail(protocol, msg)
-    local itemId = msg:getU16()
+    local itemId = msg:getU16();
+    if g_game.getClientVersion() >= 1281 then
+        local item = Item.create(itemId) -- item id
+        if item and item:getUpgradeClassification() > 0 then
+            msg:getU8()
+        end
+    end
 
     local descriptions = {}
-    for i = MarketItemDescription.First, MarketItemDescription.Last do
+    local lastAttribute = MarketItemDescription.ImbuingSlots
+    if g_game.getClientVersion() >= 1281 then
+        lastAttribute = MarketItemDescription.Last
+    end
+    for i = MarketItemDescription.First, lastAttribute do
```

Comment:
Before this modifications `MarketItemDescription.Last` was `MarketItemDescription.Weight`, not ImbuingSlots.
I'm not sure if it's intended.
https://github.com/mehah/otclient/blob/5e7d57a4d286481b634e15f92f33d3113459e180/modules/gamelib/market.lua#L126

Reply:
It wasn't. Will change the local lastAttribute back to weight when version < 1281